src/dolphin/baseline.py

```diff
@@ -0,0 +1,156 @@
+import isce3
+import numpy as np
+from numpy.typing import ArrayLike
+from opera_utils import (
+    get_cslc_orbit,
+    get_lonlat_grid,
+    get_radar_wavelength,
+)
+
+from dolphin._types import Filename
+
+
+def compute(
+    llh: ArrayLike,
+    ref_pos: ArrayLike,
+    sec_pos: ArrayLike,
+    ref_rng: float,
+    sec_rng: float,
+    ref_vel: ArrayLike,
+    ell: isce3.core.Ellipsoid,
+):
+    """Compute the perpendicular baseline at a given geographic position.
+
+    Parameters
+    ----------
+    llh : ArrayLike
+        Lon/Lat/Height vector specifying the target position.
+    ref_pos : ArrayLike
+        Reference position vector (x, y, z) in ECEF coordinates.
+    sec_pos : ArrayLike
+        Secondary position vector (x, y, z) in ECEF coordinates.
+    ref_rng : float
+        Range from the reference satellite to the target.
+    sec_rng : float
+        Range from the secondary satellite to the target.
+    ref_vel : ArrayLike
+        Velocity vector (vx, vy, vz) of the reference satellite in ECEF coordinates.
+    ell : isce3.core.Ellipsoid
+        Ellipsoid for the target surface.
+
+    Returns
+    -------
+    float
+        Perpendicular baseline, in meters.
+
+    """
+    # Difference in position between the two passes
+    baseline = np.linalg.norm(sec_pos - ref_pos)
+
+    # Compute angle between LOS vector and baseline vector
+    # via the law of cosines
+    costheta = (ref_rng**2 + baseline**2 - sec_rng**2) / (2 * ref_rng * baseline)
+
+    sintheta = np.sqrt(1 - costheta**2)
+    perp = baseline * sintheta
+    # par = baseline * costheta
```

Comment:
```suggestion
```

Reply:
I like this as a demonstration of what the other trig component gives, even though the result isn't used here. If I wasn't familiar with the algorithm my first question would be what the cos-theta component would be.

Reply:
Ok gotcha, that's true that it's helpful for that.
How about if we make ie...
```  python
    # parallel_baseline = baseline * cosine_theta

```
 (I updated the comment above)